Documents/Documents/Code/Controllers/Documents/ASCDocumentsViewController.swift

```diff
@@ -704,7 +704,7 @@ class ASCDocumentsViewController: ASCBaseTableViewController, UIGestureRecognize
     }
 
     private func createAddBarButton() -> UIBarButtonItem? {
-        guard provider?.allowAdd(toFolder: folder) == true else { return nil }
+        guard ((provider?.allowAdd(toFolder: folder)) != nil) && folder?.rootFolderType != .onlyofficeRoomArchived else { return nil }
```

Comment:
"No, because the appearance of the create room/folder button and the enable/disable state of the sortSelectBarButton button depend on the result in the allowAdd(toFolder:) method. It might be advisable to create a separate method, allowArchiveMenuAction, to split the logic.